internal/apiserver/visualization/pdf.go

```diff
@@ -2,36 +2,92 @@ package visualization
 
 import (
 	"fmt"
-	"strconv"
+	"strings"
 
 	"github.com/jung-kurt/gofpdf"
-
-	"github.com/slovak-egov/einvoice/internal/entity"
+	"github.com/lestrrat-go/libxml2"
+	"github.com/lestrrat-go/libxml2/clib"
+	"github.com/lestrrat-go/libxml2/types"
 )
 
-func Generate(inv *entity.Invoice) *File {
-	pdf := gofpdf.New("P", "mm", "A4", "")
+var lineHeight float64 = 5
+var tabSize = 4
+var font = "Arial"
+var pageHeight float64
 
-	pdf.AddPage()
+func generateLines(n types.Node, level int, pdf *gofpdf.Fpdf) error {
+	if n.NodeType() == clib.ElementNode {
+		if pdf.GetX() > pageHeight {
+			pdf.AddPage()
+		}
+
+		for i := 0; i < tabSize*(level-1); i++ {
+			pdf.Write(lineHeight, " ")
+		}
+
+		nodeNameParts := strings.Split(n.NodeName(), ":")
+		name := nodeNameParts[len(nodeNameParts)-1]
+		value := ""
+		if child, err := n.FirstChild(); err == nil && child.NodeType() == clib.TextNode {
+			value = strings.TrimSpace(child.TextContent())
+		}
+
+		pdf.SetTextColor(186, 24, 24)
+		pdf.SetFontStyle("B")
+		pdf.Write(lineHeight, name)
+		pdf.SetFontStyle("")
+		pdf.SetTextColor(0, 0, 0)
+
+		if value != "" {
+			pdf.Write(lineHeight, ": "+value)
+		}
 
-	pdf.SetFont("Arial", "B", 30)
-	pdf.Text(30, 30, "Invoice "+strconv.Itoa(inv.Id))
+		pdf.SetTextColor(191, 143, 31)
```

Comment:
comment all lines where changing color/font and explain why we change it and to what color